sw/device/lib/aes.h

```diff
@@ -0,0 +1,112 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef _F_LIB_AES_H__
+#define _F_LIB_AES_H__
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+/**
+ * Supported AES modes: encode or decode.
+ */
+typedef enum aes_mode { kAesEnc = 0, kAesDec = 1 } aes_mode_t;
```

Comment:
To me, "mode" in the context of AES normally implies something like ECB, CBC, CTR, etc - not so much encrypt vs decrypt.  But I do not have a good suggestion for what to call this instead of "mode" - maybe "operation" but I'm not in love with that either, if others have suggestions I would be curious.

If you imagine you will extend this code to have kAesEcbEnc, kAesEcbDec, and so forth, then combining mode and "operation" would be fine for me, though I'm not sure exactly how you expect to implement the crypto modes in the hardware.

Reply:
I thought about this too. Once we have other modes in hardware, this will change here as well.

Reply:
If you do ultimately end up keeping this enum as just encrypt/decrypt, then I would suggest (as @tjaychen also agreed) using "operation" because at least for me, the common usage of "mode" in this sort of block would be for the enum you use to select ECB/CBC/CTR, so anyone accustomed to the "normal" way would find it confusing.  But obviously this is going to get reworked at least one more time so we can do it then once we know.